src/gameHandlerClass.ts

```diff
@@ -0,0 +1,42 @@
+import { Player } from "./playerClass";
+import { Pokemon } from "./pokemonClass";
+
+export class GameHandler {
+  constructor(private _playerOne: Player, private _playerTwo: Player) {}
+
+  _currentPlayer: Player = this._playerOne;
+
+  getCurrentPlayer(): Player {
+    return this._currentPlayer;
+  }
+
+  getOpponentPlayer(): Player {
+    if (this._currentPlayer === this._playerOne) return this._playerTwo;
+    return this._playerOne;
+  }
+
+  isGameFinished(): boolean {
+    const firstPlayerPokemons: Pokemon[] = this._playerOne.pokemons;
+    const secondPlayerPokemons: Pokemon[] = this._playerTwo.pokemons;
+    const allPokemons = firstPlayerPokemons.concat(secondPlayerPokemons);
+    const allHPs = allPokemons.map((pokemon) => pokemon._currentHP);
+    if (!allHPs.includes(0)) {
+      return false;
+    } else {
+      return true;
+    }
```

Comment:
this logic says that if there exists at least ONE pokemon that has 0 hp - then the game is over - it should be different. If all three pokemons of one of players have 0 hp - then the game should be finished.

(`includes` method checks if there exists **at least one such element**)

Reply:
also - you shouldn't access private property "_currentHP" :) you should use "isAlive()" method

Reply:
Also you can use ready getter from player class  `this._playerOne.alivePokemons` and check if the list is empty